packages/pyroscope-flamegraph/src/convert/convertPprofToProfile.ts

```diff
@@ -0,0 +1,141 @@
+/* eslint-disable import/prefer-default-export */
+import type { Profile } from '@pyroscope/models/src';
+
+// import {Profile as PProfProfile} from 'pprof/proto';
+
+import { perftools } from 'pprof/proto/profile';
+import { string } from 'zod';
+import { deltaDiffWrapperReverse } from '../FlameGraph/decode';
+
+type node = {
+  name: string;
+  value: number;
+  total: number;
+  children: object;
+};
+
+export function convertPprofToProfile(
+  pprofBuf: Uint8Array,
+  selectedSampleType: string
+): Profile {
+  const resultFlamebearer = {
+    numTicks: 0,
+    maxSelf: 0,
+    names: [] as string[],
+    levels: [] as number[][],
+  };
+
+  const pprofProfile = perftools.profiles.Profile.decode(pprofBuf);
+
+  let sampleTypeIndex = -1;
+  for (let i = 0; i < pprofProfile.sampleType.length; i++) {
+    if (
+      pprofProfile.stringTable[pprofProfile.sampleType[i].type] ===
+      selectedSampleType
+    ) {
+      sampleTypeIndex = i;
+      break;
+    }
+  }
+
+  if (sampleTypeIndex === -1) {
+    throw 'sample type is not found in the profile';
```

Comment:
Not a big fan of throwing errors, we could return undefined so that forces people using typescript to handle `Profile | undefined`. 

Or if we are going to go with the error throwing version, to document that vehaviour with [tsdoc](https://tsdoc.org/pages/tags/example/).